src/browser/BrowserService.cpp

```diff
@@ -598,13 +597,42 @@ BrowserService::searchEntries(const QSharedPointer<Database>& db, const QString&
         if ((entryQUrl.port() > 0 && entryQUrl.port() != qUrl.port())
             || (browserSettings()->matchUrlScheme() && !entryScheme.isEmpty()
                 && entryScheme.compare(qUrl.scheme()) != 0)) {
-            continue;
+            return false;
         }
 
         // Filter to match hostname in URL field
         if ((!entryUrl.isEmpty() && hostname.contains(entryUrl))
             || (matchUrlScheme(entryUrl) && hostname.endsWith(entryQUrl.host()))) {
-            entries.append(entry);
+            return true;
+        }
+        return false;
+    };
+
+    // Search host from URL fields
+    for (auto* entry : EntrySearcher().search(baseDomain(hostname), rootGroup)) {
+        if (!handleURL(entry->url())) {
+            continue;
+        }
+
+        entries.append(entry);
+    }
+
+    // Search for additional URL's starting with KP2A_URL
+    for (const auto group : rootGroup->groupsRecursive(true)) {
+        if (group->isRecycled() || !group->resolveSearchingEnabled()) {
+            continue;
+        }
+
+        for (auto* entry : group->entries()) {
+            if (entry->isRecycled() || !entry->attributes()->keys().contains("KP2A_URL")) {
+                continue;
+            }
+
+            for (const auto& key : entry->attributes()->keys()) {
+                if (key.startsWith("KP2A_URL") && handleURL(entry->attributes()->value(key))) {
+                    entries.append(entry);
+                }
+            }
         }
     }
```

Comment:
These loops can be combined. You are effectively recursing through the entire database twice. I wouldn't use the entry searcher in this case.

Reply:
I agree. Two loops are not necessary. The reasong this was left this way was that EntrySearcher didn't support such regex based search based on attribute's name. I'll clean this up.

And writing some extra non-GUI tests would be also a nice addition.

Reply:
Changes made.